src/dbnode/storage/index/convert/convert.go

```diff
@@ -188,6 +190,76 @@ func FromSeriesIDAndTagIter(id ident.ID, tags ident.TagIterator) (doc.Metadata,
 	return d, nil
 }
 
+// FromSeriesIDAndEncodedTags converts the provided series id and encoded tags into a doc.Metadata.
+func FromSeriesIDAndEncodedTags(id ident.ID, encodedTags ts.EncodedTags) (doc.Metadata, error) {
```

Comment:
Please change `ident.ID` to `ident.BytesID`. I've just realized that using an interface here will force an allocation on some of the code paths where we will be using this function.

Reply:
https://github.com/m3db/m3/pull/3087/commits/8b725f006584dce7951b0a8d4a83b64e05c644bd